src/resolve/resolved-conf.c

```diff
@@ -52,7 +53,12 @@ static int manager_add_dns_server_by_string(Manager *m, DnsServerType type, cons
                 return 0;
         }
 
-        return dns_server_new(m, NULL, type, NULL, family, &address, ifindex);
+        r = dns_server_new(m, NULL, type, NULL, family, &address, ifindex, server_name);
+        if (r < 0)
+            return r;
+
+        TAKE_PTR(server_name);
```

Comment:
hmm, we usually don't donate strings like that. I don't think we should here either, to keep this more inline with out usual codebase... i.e. just pass server_name as it is, and let dns_server_new() copy it

Reply:
I can admit I'm not really familiar with all of it.
I tried to fix it as good as I can :)